mlir/utils/buildbot/mlir-rocm-mi200/Dockerfile

```diff
@@ -17,13 +17,9 @@ FROM rocm/mlir:rocm6.2-latest
 RUN apt-get update; \
     apt-get install -y dumb-init;
 
-# LTS releases often bundle obsolete pip versions that cannot access newest
-# Linux binary wheels.
-RUN python3 -m pip install --upgrade pip && python3 -m pip install --ignore-installed PyYAML
-
-# Refer to mlir/lib/Bindings/Python/requirements.txt. Listed explicitly here
-# and version pinned for consistency as this is a bot.
-RUN python3 -m pip install numpy 'pybind11>=2.8.0' dataclasses
+# Accounts for PyYAML, numpy, pybind11, ml_dtypes
+ADD "https://raw.githubusercontent.com/llvm/llvm-project/main/mlir/python/requirements.txt" requirements.txt
```

Comment:
A new test from upstream requires the python module ml_dtypes, which we don't install anywhere.  Rather than add it to the explicit list, let's use the requirements.txt file that changed to include it.

Grabbing from upstream rather than rocMLIR because the buildbot is sent upstream code to build and we need it to keep up.  Next time they add a module, we just have to kill and restart the buildbot and it'll pick it up (I think).